src/components/client/core/Slider/Slider.test.tsx

```diff
@@ -0,0 +1,54 @@
+import { describe, it } from "@jest/globals";
+import { render, screen } from "@testing-library/react";
+
+import { Slider } from "components/client";
+
+import type { SliderMarkerRecord } from "components/client";
+
+const MARKERS: SliderMarkerRecord[] = [
+  { id: "marker-1", value: 25, label: "25%" },
+  { id: "marker-2", value: 50, label: "50%" },
+  { id: "marker-3", value: 75, label: "75%" },
+];
+
+const MockSlider = () => (
+  <Slider min={0} max={100} maxW="md" label="Slider Label" markers={MARKERS} />
+);
+
+describe("Slider", () => {
+  const { ResizeObserver } = window;
+
+  beforeEach(() => {
+    // @ts-ignore TS2790: The operand of a 'delete' operator must be optional.
+    delete window.ResizeObserver;
+    window.ResizeObserver = jest.fn().mockImplementation(() => ({
+      observe: jest.fn(),
+      unobserve: jest.fn(),
+      disconnect: jest.fn(),
+    }));
+  });
+
+  afterEach(() => {
+    window.ResizeObserver = ResizeObserver;
+    jest.restoreAllMocks();
+  });
```

Comment:
No need to change anything in this PR, but we may need to mock the `ResizeObserver` API in multiple tests, in which case we can inject a shared mock to all the tests. I have examples of this (e.g. with `IntersectionObserver`) in `mirageswap-app` repo (`src/__mocks__`). Something to keep in mind just if this needs to be mocked in other tests down the road

Reply:
Ok sounds good! Will take a look!